reproducible/README.md

```diff
@@ -0,0 +1,343 @@
+
+The script "build-oe-nix-image.sh" is a thin wrapper. It takes no arguments, and just passes the needed arguments
+to docker build.  If it needs to be modified, a new script should be created.
+
+Dockerfile.nix prepares the build image by:
+
+- adding via apt minimum packages for the outer image. This includes a text editor (vim) and ELF patch utility (patchelf).
+
+- sets up nit configuration.
+
+- diables apt.
+
+- creates the build user and directories, groups etc.
+
+- creates a nix expression for building openenclave via (shell.nix).
+
+- installs nix as the build user
+
+- calls the prep-nix-build.sh script which sets up the git clone of nix pkgs and preinstalls the package
+  derivations needed for building openenclave into the nix store. The search process for dependencies in
+  nix is to refer to the nix store first, before looking at nixpkgs or contacting the cache if enabled. While
+  we cannot select package versions from the expression, but must depend on the nixpkgs repo to provide the
+  correct versions, we can prime the nix store to provide the exact derivation we wish before referring to the repo.
+
+  These derivations have been manually chosen.  This also allows us to set package priorities which is required
+  between packages that provide overlapping contents, for example clang_7 and llvm_7.
+
+- adds a build.env file which contains environment values to control the build.  This is not part of the 
+  - REV is the git revision (commit or tag) to pull for the build.
+  - SHA is the sha256 hash of the tar file downloaded from git.
+  - DO_CHECK enables ctests as part of the build.
+  - DEB_SHA specifies the expected result sha25sum for the .deb file produced. An error will be thrown if the 
+    sha is incorrect. If not set, the sha is not checked.
+
+- adds needed files for ctest to the .nix_libs directory, which will be referred to by ctest. Nix expressions
+  cannot refer to libraries outside the minimum environment and explicit inputs, so these libraries are needed
+  to use sgx (and eventually TrustZone) facilties.
+
+- sets the build user as the user. Among other things, this means the build image when run cannot be 
+  modified by the addition of apt packages.
+
+The resulting image is named openenclave-build. It can (and should) be generated once and pushed into a common 
+repo.
+
+### build_openenclave.sh
+
+Runs the image produced by docker build, with arguments resolved.
+The output of the process is a .deb and a .nar file containing an exported nix store for use in developing 
+openenclave sdk apps using nix.
+
+
+### workflow
+
+```
+   ./build-oe-nix-image.sh
+```
+will build the nix docker image. This can be pushed:
+```
+   docker push openenclave-build
+```
+and later pulled, or just used locally.
+
+To build the openenclave sdk 
+```
+   ./build_openenclave.sh -o ./result
+```
+will build the oe deb and nix-store and place it in ./result. By default it will run ctests. If the
```

Comment:
At the end of the build, couldn't find the debian package in `./result` on the host or in `/output` in the container. 
The debian package is created under the nix-store in the container, but probably isn't copied to `/output` ?

```
azureuser@62548e384ec9:~$ find / -type f -name "open-enclave_0.12.0_amd64.deb" 
/nix/store/0kjcfn87x6ai26nk8s7scq8wc5mqikcg-openenclave-sdk/open-enclave_0.12.0_amd64.deb
/nix/store/0kjcfn87x6ai26nk8s7scq8wc5mqikcg-openenclave-sdk/_CPack_Packages/Linux/DEB/open-enclave_0.12.0_amd64.deb
```

Reply:
@yakman2020, can you clarify the expectation on the location of the Debian package?

Reply:
The package goes into /output in the container, or the directory specified by -o on the host.

Reply:
brcamp@nixbed:~/Projects/openenclave/reproducible$ ls ~/results 
build  open-enclave_0.12.0_amd64.deb

Reply:
Try 
```
git reset --hard HEAD~1
git pull
```

I've been force pushing into the PR commit.

Reply:
And , by the way, thank you for your feedback.

Reply:
changed to ~/result which is an absolute path. Relative paths are not allowed.